pyg_dataset.py

```diff
@@ -54,12 +55,11 @@ def processed_file_names(self):
         That means that if you want to reprocess the data, you need to delete
         the processed files and reimport the dataset.
         """
-        # TODO: Automatically detect changes in the dataset.
-        #       We could come up with a namig scheme that will differentiate
-        #       which graph families (and/or sizes) and features were used to
-        #       generate the dataset. This way, we could detect changes and
-        #       reprocess the dataset when needed.
-        return ["data.pt"]
+        dataset_props = json.dumps([self.loader.selection, self.features])
+        sha256_hash = hashlib.sha256(dataset_props.encode("utf-8")).digest()
+        hash_string = base64.urlsafe_b64encode(sha256_hash).decode("utf-8")[:10]
+
+        return [f"data_{hash_string}.pt"]
```

Comment:
Makes a hash of selected graph sizes and features, and generates a unique file name. If the same parameters are given in another run, and if the file exists, it will be loaded. Otherwise, the dataset will be automatically regenerated.